src/cache/mod.rs

```diff
@@ -22,18 +23,38 @@ where
 {
     items: RwLock<BTreeMap<T, Item<V>>>,
     item_duration: Option<Duration>,
+    capacity: usize,
 }
 
 #[allow(clippy::len_without_is_empty)]
 impl<T, V> Cache<T, V>
 where
     T: Ord + Hash,
 {
-    ///
-    pub fn new(item_duration: Option<Duration>) -> Self {
-        Cache {
+    /// Creating capacity based `Cache`.
+    pub fn with_capacity(capacity: usize) -> Self {
+        Self {
+            items: RwLock::new(BTreeMap::new()),
+            item_duration: None,
```

Comment:
to confirm that will this mean to set expiry as to Now, which means the item is `expired` immediately?

Reply:
No.

Condition to consider an item expired as follows:

```rust
pub fn expired(&self) -> bool {
        self.expiry
            .map(|expiry| expiry < Instant::now())
            .unwrap_or(false)
    }
```